core/store/orm/orm.go

```diff
@@ -1046,6 +1049,12 @@ func (orm *ORM) UpdateBridgeType(bt *models.BridgeType, btr *models.BridgeTypeRe
 // CreateInitiator saves the initiator.
 func (orm *ORM) CreateInitiator(initr *models.Initiator) error {
 	orm.MustEnsureAdvisoryLock()
+	if initr.JobSpecID == nil {
+		// FIXME: This hangs forever if we don't check this here and the
+		// supplied initiator does not have a JobSpecID set.
+		// I do not know why. Seems to be something going wrong inside gorm
+		return errors.New("requires job spec ID")
```

Comment:
This seems like a program invariant, does this error get logged at least?

Reply:
No but it looks like it's only called from tests anyway.

Reply:
This function can't move because it needs access to the `db` object which is unexported. So it will have to stay where it is. I'll add some logging.